tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -2909,6 +2909,9 @@ void CopyMatMulAttributes(const NodeDef& matmul, NodeDef* fused_matmul,
     auto& activation_attr = activation->attr();
     (*attr)["leakyrelu_alpha"] = activation_attr.at("alpha");
   }
+  if (IsMKLEnabled()) {
+    (*attr)["_input_shapes"] = src_attr.at("_input_shapes");
+  }
```

Comment:
The crash can be fixed with this:
```suggestion
  if (IsMKLEnabled()) {
    auto input_shapes = src_attr.find("_input_shapes");
    if (input_shapes != src_attr.end()) {
      (*attr)["_input_shapes"] = input_shapes->second;
    }
  }
```
But now I got lots of minor numerical errors instead. Will see how much we'll need to tweak the tolerance and check with @cantonios whether he thinks they are reasonable.
```
[ RUN      ] RemapperTest.FuseConv3DWithBiasAndAddN
tensorflow/core/framework/tensor_testutil.cc:184: Failure
Value of: IsClose(Tx[i], Ty[i], typed_atol, typed_rtol)
  Actual: false (17720488 not close to 17720490)
Expected: true
i = 918911 Tx[i] = 17720488 Ty[i] = 17720490
...
third_party/tensorflow/core/framework/tensor_testutil.cc:184: Failure
Value of: IsClose(Tx[i], Ty[i], typed_atol, typed_rtol)
  Actual: false (17745482 not close to 17745484)
Expected: true
i = 920191 Tx[i] = 17745482 Ty[i] = 17745484
...
```

Reply:
Yes, Thank you for the fix. We have to add the fix for CopyBatchMatMulAttributes() as well. I will submit the patch shortly to add that in
.

Reply:
The test actually passes now after I apply the same fix to `CopyBatchMatMulAttributes` as well. I'll rerun the internal presubmits and will get back here if there are more errors.

Reply:
Ok thank you